src/main/java/com/minecraftcorp/lift/bukkit/listener/PlayerListener.java

```diff
@@ -164,6 +164,35 @@ public void onPlayerMove(PlayerMoveEvent event) {
 		player.setVelocity(pushBack);
 	}
 
+	@EventHandler
+	public void onPlayerChangedWorldEvent(PlayerChangedWorldEvent event) {
+		Player player = event.getPlayer();
+		Location playerLocation = player.getLocation();
+
+		boolean playerInNoLift = plugin.isInNoLift(player.getUniqueId());
+
+		if (playerInNoLift) {
+			return;
+		}
+
+		BukkitElevator elevator = plugin.getActiveLifts()
+				.stream()
+				.findFirst()
+				.orElse(null);
+
+		if (elevator == null) {
+			return;
+		}
+
+		plugin.logDebug(player.getName() + " changed a world while lifting.");
+
+		elevator.removePassengers(Collections.singletonList(player));
+		elevator.removeFreezers(Collections.singletonList(player));
+
+		player.teleport(playerLocation, PlayerTeleportEvent.TeleportCause.PLUGIN);
```

Comment:
Could you help me why we need to teleport the player here?

Reply:
This is the code I wrote before implementing moving the passengers/freezers, to return to the original destination after being moved to the elevator's destination coordinates.

However, I have now removed this code, as I have confirmed that it works correctly even if it is removed.